File: Models/RiskReport.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TransactionAPI.Models;

public enum RiskLevel
{
    Low,
    Medium,
    High
}

public class RiskReport
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int ReportID { get; set; }

    [Required]
    [MaxLength(50)]
    public string TransactionID { get; set; } = string.Empty;

    [Required]
    public RiskLevel RiskLevel { get; set; }

    [MaxLength(500)]
    public string? DetectedAnomaly { get; set; }

    [MaxLength(1000)]
    public string? RecommendedMitigation { get; set; }

    [MaxLength(2000)]
    public string? Reasoning { get; set; }

    [MaxLength(500)]
    public string? TLDR { get; set; }

    // Navigation property
    [ForeignKey("TransactionID")]
    public virtual Transaction? Transaction { get; set; }
}

File: client/src/types/riskReport.ts
export type RiskLevel = 'Low' | 'Medium' | 'High';

// Helper function to convert numeric enum from .NET API to RiskLevel string
export const mapRiskLevel = (value: number | string): RiskLevel => {
  if (typeof value === 'string') {
    return value as RiskLevel;
  }
  switch (value) {
    case 0:
      return 'Low';
    case 1:
      return 'Medium';
    case 2:
      return 'High';
    default:
      return 'Low';
  }
};

export interface RiskReport {
  reportID: number;
  transactionID: string;
  riskLevel: RiskLevel | number; // Accept both string and number from API
  detectedAnomaly: string | null;
  recommendedMitigation: string | null;
  transaction?: {
    transactionID: string;
    accountID: string;
    amount: number;
    merchant: string;
    category: string;
    timestamp: string;
    location: string;
  };
}
